lib/rubocop/cop/lint/uri_regexp.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop identifies places where `URI.regexp`
+      # can be replaced by `URI::DEFAULT_PARSER.make_regexp`.
+      #
+      # @example
+      #   # bad
+      #   URI.regexp("http://example.com")
+      #
+      #   # good
+      #   URI::DEFAULT_PARSER.make_regexp("http://example.com")
+      #
+      class UriRegexp < Cop
+        MSG = 'Use `%sURI::DEFAULT_PARSER.make_regexp%s` instead of ' \
+              '`%sURI.regexp%s`.'.freeze
+
+        def_node_matcher :uri_regexp_with_argument?, <<-PATTERN
+          (send
+            (const $_ :URI) :regexp
```

Comment:
This pattern matches `MyClass::URI.regexp('str')` also.

```bash
$ ruby-parse -e ::URI; MyClass::URI
(begin
  (const
    (cbase) :URI)
  (const
    (const nil :MyClass) :URI))
```


Could you change this matcher to use `{nil cbase}`?

Reply:
Sure 💦  I fixed by ced09d3.